### DISCUSSION TITLE
Adding a switch (or another better name) operator

### DISCUSSION THREAD

**Discussion by LcsGa** [ReactiveX/rxjs#7011] [Ideas / Feature request]
Sometimes I need to subscribe to an observable in order to, for example, get a userId.
Then, using that id, I could send an http request and for that purpose I often end up using a switchMap.
The thing is that if I want to send more than one http requests, with all or some of them needing the userId, I would have to "bother" myself by adding a map to each switchMap.

And this is specifically for this purpose that I would love to have an operator that would just call for another observable but like tap, would keep the value unchanged.

```ts
this.user$.pipe(
  take(1),
  map(user => user.id),
  switch(userId => this.someRequest(userId)),
  switch(userId => this.someOtherRequest(userId))
).subscribe();
```

**Comment by martinsik** [comment]
You could use `concat` like this (or `forkJoin` depending on what you want to do):

```
this.user$.pipe(
  take(1),
  map(user => user.id),
  switchMap(userId => concat(this.someRequest(userId), this.someOtherRequest(userId))),
).subscribe();
```

This will keep the order of calls and if you don't want to propagete their responses to your observer you could add `toArray(), map(() => userId)`.
I think this is not a very common use-case however, usually you want to do something with the responses and if not, you can easily use what I posted above.

**Reply by LcsGa** [reply]
Hi! Thanks for your reply! In fact we do need to do this on some complex forms we need to update.
Our forms are often using many "blocks" and we do want to update all of them, at the same time, on a register button.
Since each block is updated depending on the same user, that is why we need to use the same id everywhere, but since we do not want to send unnecessary request, between each switchMap we used to do, we often had other observables like a filter for example, in that case, don't you think a simple switch could do the trick? And that switch would in fact be the combination of a switchMap with a map returning the initial value used by the switchMap
